### PR TITLE
fix(expo-video): NaN current time when source is not provided

### PR DESCRIPTION
**Pull request by AlirezaHadjar**
# Why

When the video source is not provided, `player.currentTime` returns `NaN` which is not ideal (`player.duration` returns `0`). Android is working fine and returns `0` and only iOS is broken.

# How
I put an `if condition` and check if the value is `nan` then return `0`

# Test Plan
Here is how this scenario can be replicated:
Update:
```ts
  const [currentSource, setCurrentSource] = React.useState(videoSources[0]);
```
to: 
```ts
  const [currentSource, setCurrentSource] = React.useState(null);
```

Now if you log the `player.currentTime`, it's `NaN`. The fix that I provided changes it to `0`

# Checklist

- [ ] Documentation is up to date to reflect these changes (eg: https://docs.expo.dev and README.md).
- [ ] Conforms with the [Documentation Writing Style Guide](https://github.com/expo/expo/blob/main/guides/Expo%20Documentation%20Writing%20Style%20Guide.md)
- [x] This diff will work correctly for `npx expo prebuild` & EAS Build (eg: updated a module plugin).

### CODE REVIEW

**Comment by hirbod** [comment]
Besides `currentTime` being `NaN`, I encountered issues with reliably obtaining the duration. The `player.duration` inside the hook callback is `NaN` the first time you access it. Even adding listeners like `statusChange` is no guarantee.

I had to resort to using `requestAnimationFrame` as a workaround, because the instance might still be NaN.
I tried to useEffect with `player.duration` as dependency, which worked on iOS but wouldn't cause a re-render on Android. Likely a coincidence, but interesting observation. 

```ts
useEffect(() => {
  const listener = player.addListener('statusChange', (status) => {
    if (status === 'readyToPlay' || !Number.isNaN(player.duration)) {
      requestAnimationFrame(() => {
        console.log('duration', player.duration)
        setDuration(player.duration)
      })
    }
  })

  return () => {
    listener.remove()
  }
}, [player, setDuration])
```

I believe we need dedicated events for `currentTime` and `duration`, or we should ensure that events like `statusChange` include them as metadata.

**Comment by AlirezaHadjar** [comment]
@hirbod Yeah, I totally agree. having an event specifically for the progress is very useful. I also need this event because updating a custom slider with the video's current time is not ideal. I have to use `useInterval` which sometimes causes it to get unsynced or left behind. [RNTP](https://rntp.dev/docs/api/hooks#useprogress) handles this beautifully. If we have an event for progress we can create a similar `useProgress` hook.

But to begin with, don't you agree that neither `currentTime` nor `duration` should ever be `NaN`? They should at least be `0`. I'm thinking of adding the same `time.isNaN` for the duration until we have a better way.

**Comment by hirbod** [comment]
Yes, NaN is weird, but at least it helps indicate that it's 'undetermined' to begin with. But 0 seems more natural. By the way, accessing currentTime on Android is a blocking operation, so your interval approach will likely not work on Android at the moment.

**Comment by AlirezaHadjar** [comment]
Yeah, I noticed that it slows down on Android when it's called within an interval. Let's wait for feedback from an Expo team member to see what they think is the best solution for now 🙂.

File: packages/expo-video/CHANGELOG.md [hunk 1]
```diff
@@ -8,6 +8,7 @@
 
 - [iOS] Add live indicator in the now playing info. ([#30629](https://github.com/expo/expo/pull/30629) by [@justjoostnl](https://github.com/justjoostnl))
 - [iOS] Add 'allowsExternalPlayback' property to control the AirPlay to stream video or mirror the screen. ([#30436](https://github.com/expo/expo/pull/30436) by [@adsalihac](https://github.com/adsalihac))
+- [iOS] Fix 'player.currentTime' NaN when source is not provided. ([#31011](https://github.com/expo/expo/pull/31011) by [@AlirezaHadjar](https://github.com/AlirezaHadjar))
```
Comment (tsapeta) [inline packages/expo-video/CHANGELOG.md:11]:
```suggestion
- [iOS] Fixed `player.currentTime` being `NaN` when source is not provided. ([#31011](https://github.com/expo/expo/pull/31011) by [@AlirezaHadjar](https://github.com/AlirezaHadjar))
```

**Review by tsapeta — APPROVED** [review]

**Comment by AlirezaHadjar** [comment]
@tsapeta I can confirm that as @hirbod said, the duration will also be `NaN` when you update the source. And adding the same `duration.isNaN` will fix the issue. Should I add it to the duration as well?

**Comment by tsapeta** [comment]
@AlirezaHadjar Yes, if you could fix the duration that would be great 😉 Also I noticed that we have `currentTime` twice in the definition, could you remove the first one (which has no setter)?

**Comment by AlirezaHadjar** [comment]
Sure thing!

**Review by tsapeta — APPROVED** [review]

**Pull request merged**